engine/src/cmd/unit_csv.cpp

```diff
@@ -65,28 +65,9 @@ extern void pushMesh(std::vector<Mesh *> &mesh,
 void addShieldMesh(Unit::XML *xml, const char *filename, const float scale, int faction, class Flightgroup *fg);
 void addRapidMesh(Unit::XML *xml, const char *filename, const float scale, int faction, class Flightgroup *fg);
 
-// TODO: This is a terrible kludge. Replace with boost::json
 std::string MapToJson(std::map<std::string, std::string> unit) {
-    std::string json_string = "[\n\t{\n";
-    
-    int len = unit.size();
-    int i = 0;
-
-    for (auto const& pair : unit) {
-        boost::format new_line;
-        if(i < len-1) {
-            new_line = boost::format("\t\t\"%1%\": \"%2%\",\n") % pair.first % pair.second;
-        } else {
-            new_line = boost::format("\t\t\"%1%\": \"%2%\"\n") % pair.first % pair.second;
-        }
-        
-        i++;
-        json_string += new_line.str();
-    }
-
-    json_string += "\t}\n]\n";
-    
-    return json_string;
+    std::string return_value = boost::json::serialize(boost::json::value_from(unit));
```

Comment:
Good! However, it's only used once and is now a really trivial function. Please fold it into WriteUnit.

Reply:
OK, I can do that.

This is the actual fix. The prior implementation of this function didn't escape backslashes properly.

Reply:
Done.